extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -0,0 +1,344 @@
+  }
+
+  /**
+   * Builds a button and appends it to the container.
+   * @param {!string} label Text of the button for a11y
+   * @param {!string} className Css classname
+   * @param {!function()} action function to call when tapped
+   * @private
+   */
+  buildButton_(label, className, action) {
+    const button = this.win.document.createElement('div');
+
+    button.setAttribute('role', 'button');
+    button.setAttribute('aria-label', label);
+    button.classList.add(className);
+    button.addEventListener('click', action);
+
+    this.container_.appendChild(button);
+  }
+
+  /**
+   * Opens the lightbox-viewer with either the invocation source or
+   * the element referenced by the `id` argument.
+   * Examples:
+   *  // Opens the element tapped.
+   *  on="tap:myLightboxViewer'
+   *
+   *  // Opens the element referenced by elementId
+   *  on="tap:myLightboxViewer.open(id='<elementId>')
+   * @override
+   */
+  activate(invocation) {
+    let target = invocation.source;
+    if (invocation.args && invocation.args.id) {
+      const targetId = invocation.args.id;
+      target = this.win.document.getElementById(targetId);
+      user().assert(target,
+        'amp-lightbox-viewer.open: element with id: %s not found', targetId);
+    }
+    this.open_(target);
+  }
+
+  /**
+   * Opens the lightbox-viewer and displays the given element inside.
+   * @param {!Element} elem Element to lightbox.
+   * @private
+   */
+  open_(elem) {
+    if (this.activeElem_ == elem) {
+      return;
+    }
+
+    this.updateViewer_(elem);
+
+    this.element.style.display = 'block';
+    this.active_ = true;
+
+    this.win.document.documentElement.addEventListener(
+        'keydown', this.boundHandleKeyboardEvents_);
+  }
+
+  /**
+   * Closes the lightbox-viewer
+   * @private
+   */
+  close_() {
+    if (!this.active_) {
+      return;
+    }
+
+    this.element.style.display = 'none';
+    this.tearDownElem_(this.activeElem_);
+
+    this.activeElem_ = null;
+    this.active_ = false;
+
+    this.win.document.documentElement.removeEventListener(
+        'keydown', this.boundHandleKeyboardEvents_);
+  }
+
+  /**
+   * Opens the next element to be displayed in the lightbox.
+   * @private
+   */
+  next_() {
+    dev().assert(this.activeElem_);
+    this.manager_.getNext(this.activeElem_).then(nextElem => {
+      if (nextElem) {
+        this.updateViewer_(nextElem);
+      }
+    });
+  }
+
+  /**
+   * Opens the previous element to be displayed in the lightbox.
+   * @private
+   */
+  previous_() {
+    dev().assert(this.activeElem_);
+    this.manager_.getPrevious(this.activeElem_).then(prevElem => {
+      if (prevElem) {
+        this.updateViewer_(prevElem);
+      }
+    });
+  }
+
+  /**
+   * Updates the viewer to display the new element and tears down the old one
+   * @param {!Element} newElem
+   * @private
+   */
+  updateViewer_(newElem) {
+    const previousElem = this.activeElem_;
```

Comment:
Not a huge fun of `elem` abbreviation. All of these are obfuscated by the compiler. If all the same, could you spell out `element`?